userspace/libsinsp/logger.cpp

```diff
@@ -205,3 +271,84 @@ char* sinsp_logger::format(const char* fmt, ...)
 
 	return m_tbuf;
 }
+
+const char *sinsp_logger::encode_severity(const severity sev)
```

Comment:
I suggest that you return a `const std::string&` instead of a `const char *` and just let the client call `c_str()` on what you give them back when they need a C-style string.  That way `decode_severity` won't have to generate temporary objects to do the comparisons.

Reply:
We don't really compare strings here but convert the first ENCODE_LEN (= 8) bytes of the string/severity values to uint64_t and compare these. I'm not sure there's a nice equivalent of `strncmp` for `std::string`, so I just replaced the pointer casts with memcpy calls to avoid alignment issues (they should be optimized out to a `cmp reg, [reg]` anyway).

If you feel this is too hacky, I'm all ears.